### ISSUE TITLE
'getImageCallBack is not defined' - Exporting as PNG

### ISSUE THREAD

**Issue by jkhamler** [kevinkhill/lavacharts#218]
Hi Kevin,

I'm trying to export my charts as PNGs and have got held up on this error. Can you recommend a way forward? 

Many thanks

Jon

#### What Version?
3.0.8
#### Issue
> Please describe the issue. 
 'getImageCallBack is not defined'

#### Controller Code (chart creation code)
```
        \Lava::PieChart('TotalCallsReceived', $totalCallsReceived, [
            'events' => ['ready' => 'getImageCallBack'],
            'title' => 'Total Calls Received & Transferred',
            'is3D' => false,
        ]);

```

#### View Code

Before the closing 'head' tag:
```
    <script type="text/javascript">
        function getImageCallback(event, chart) {
            console.log(chart.getImageURI());
            // This will return in the form of "data:image/png;base64,iVBORw0KGgoAAAAUA..."
        }
    </script>
```
My chart div:
```
                <div id="total-calls-div"></div>
                {!! \Lava::render('PieChart', 'TotalCallsReceived', 'total-calls-div') !!}
```
![screen shot 2017-08-29 at 13 30 05](https://user-images.githubusercontent.com/7847437/29820966-30bb501c-8cbe-11e7-9ddb-28e418b74971.png)

**Comment by kevinkhill** [comment]
- First, please update to the most recent version
 - Second, instead of using the events to get the image-uri of the chart, you can simply add `['png' => true]` to the options for the chart. If the chart supports viewing as a png, then it will be drawn as such.

*side note*
Not all charts support PNG output, so a browse through the chart classes, [such as AreaCharts](https://github.com/kevinkhill/lavacharts/blob/3.1/src/Charts/AreaChart.php) will show if they `use \Khill\Lavacharts\Support\Traits\PngRenderableTrait`

**Comment by jkhamler** [comment]
That's really helpful, thanks Kev. Just one thing - I've got a composer error when I try to update:

```
  Problem 1
    - The requested package khill/lavacharts 3.1 exists as khill/lavacharts[3.0.8] but these are rejected by your constraint.
```

```
    "khill/lavacharts": "~3.1",
```

**Comment by kevinkhill** [comment]
We resolved this over email, right?

**Comment by kevinkhill** [comment]
Closing due staleness and lack of communication. Please open a new issue of you still need assistance, as I am happy to help. ☺️

**Comment by obayesshelton** [comment]
@kevinkhill I am using the following:

```
\Lava::LineChart('MyChart', $population, [
    'png' => true
]);
```

What method will return me the image string? I cannot seem to find how we generate the image URI or base64 encoded string

**Comment by kevinkhill** [comment]
If you [take a look at the wiki](https://github.com/kevinkhill/lavacharts/wiki) there are examples of how to get the base64 string

**Comment by obayesshelton** [comment]
Hey @kevinkhill thanks for the reply.

In - https://github.com/kevinkhill/lavacharts/wiki/Getting-a-chart-as-an-image-3.1.x you do not use a call back so I was curious if I could pass the base64 string into blade as a variable?

I'm trying to create a PNG of the graph pass it into blade which then will be used to create a PDF using Laravel DomPDF

**Comment by kevinkhill** [comment]
I'm sorry, I guess I should explain it better in the wiki, but the only way
for Lavacharts to work, is to have Javascript execution. Whether in the
browser, or using PhantomJs or NightmareJs on the server.

Since Google Charts is a Javascript library, Lavacharts just generates the
needed code for chart generation, but doesn't actually generate the charts.

This comes up often, and I wish I had a more elegant solution, but as it
stands now, getting a direct png output to the page is not possible.

Using the png = true option just replaces the svg chart with the png chart,
during the page load, in the browser.

**Comment by obayesshelton** [comment]
Ahhh perfect, thanks is fine 👍

_I don't envy the job of being a maintainer_

**Comment by kevinkhill** [comment]
I wish I had time to "maintain" my library. Babies don't give up much free time for hobby coding 😂

I should just make a big disclaimer on the main Readme. This is the most commonly asked question / issue raised

**Comment by obayesshelton** [comment]
haha yes, good look and thanks!
